src/Simplex/Chat/Store/Profiles.hs

```diff
@@ -524,177 +530,262 @@ updateUserAddressAutoAccept db user@User {userId} autoAccept = do
+          WHERE server_operator_id = ?
+        |]
+        (tradeName, legalName, T.intercalate "," serverDomains, enabled, storage, proxy, operatorId)
+    insertOperator :: NewServerOperator -> IO ServerOperator
+    insertOperator op@ServerOperator {operatorTag, tradeName, legalName, serverDomains, enabled, roles = ServerRoles {storage, proxy}} = do
+      DB.execute
+        db
+        [sql|
+          INSERT INTO server_operators
+            (server_operator_tag, trade_name, legal_name, server_domains, enabled, role_storage, role_proxy)
+          VALUES (?,?,?,?,?,?,?)
+        |]
+        (operatorTag, tradeName, legalName, T.intercalate "," serverDomains, enabled, storage, proxy)
+      opId <- insertedRowId db
+      pure op {operatorId = DBEntityId opId}
+    autoAcceptConditions op UsageConditions {conditionsCommit} =
+      acceptConditions_ db op conditionsCommit Nothing
+        $> op {conditionsAcceptance = CAAccepted Nothing}
+
+getServerOperators_ :: DB.Connection -> IO [ServerOperator]
+getServerOperators_ db =
+  map toOperator
+    <$> DB.query_
+      db
+      [sql|
+        SELECT server_operator_id, server_operator_tag, trade_name, legal_name,
+          server_domains, enabled, role_storage, role_proxy
+        FROM server_operators
+      |]
+  where
+    toOperator (operatorId, operatorTag, tradeName, legalName, domains, enabled, storage, proxy) =
+      ServerOperator
+        { operatorId,
+          operatorTag,
+          tradeName,
+          legalName,
+          serverDomains = T.splitOn "," domains,
+          conditionsAcceptance = CARequired Nothing,
+          enabled,
+          roles = ServerRoles {storage, proxy}
+        }
+
+getOperatorConditions_ :: DB.Connection -> ServerOperator -> UsageConditions -> Maybe UsageConditions -> UTCTime -> IO ConditionsAcceptance
+getOperatorConditions_ db ServerOperator {operatorId} UsageConditions {conditionsCommit = currentCommit, createdAt, notifiedAt} latestAcceptedConds_ now = do
+  case latestAcceptedConds_ of
+    Nothing -> pure $ CARequired Nothing -- no conditions accepted by any operator
+    Just UsageConditions {conditionsCommit = latestAcceptedCommit} -> do
+      operatorAcceptedConds_ <-
+        maybeFirstRow id $
+          DB.query
+            db
+            [sql|
+              SELECT conditions_commit, accepted_at
+              FROM operator_usage_conditions
+              WHERE server_operator_id = ?
+              ORDER BY operator_usage_conditions_id DESC
+              LIMIT 1
+            |]
+            (Only operatorId)
+      pure $ case operatorAcceptedConds_ of
+        Just (operatorCommit, acceptedAt_)
+          | operatorCommit /= latestAcceptedCommit -> CARequired Nothing -- TODO should we consider this operator disabled?
+          | currentCommit /= latestAcceptedCommit -> CARequired $ conditionsRequiredOrDeadline createdAt (fromMaybe now notifiedAt)
+          | otherwise -> CAAccepted acceptedAt_
```

Comment:
this trying to save up on one extra branch and using guards instead of if else leads to overall less understandable code. instead of doing direct check

latestAcceptedCommit == currentCommit && operatorCommit == latestAcceptedCommit -> accepted

you now have to read all guards above and negate them to try to understand if this is valid logic:
"""otherwise is "!(operatorCommit /= latestAcceptedCommit) && !(currentCommit /= latestAcceptedCommit)" (and would be more conditions if other guards were above) so it's "(operatorCommit == latestAcceptedCommit) && (currentCommit == latestAcceptedCommit)" so it seems to be correct"""

so instead of going over logic branch by branch and understanding simple conditions one at a time, you now have to go over all cases and do some perverse logic manipulations. otherwise may be good for default (or at best primitive if-else), but here it's not default - if anything accepted is the furthest from default - and this construction simply doesn't fit this case well and forces to understand more complex logic

Reply:
simpler logical expressions are simpler to understand. These branches define the "original" logic that should be applied here, not some simplified logic, and it is simpler than using the matrix.

The logic is a sequence of decision steps:
1) if operator commit is different from latest accepted, it has to be accepted now (Required nothing)
2) or if all operators commit is different from current, then it will be re-accepted at some deadline
3) otherwise it is accepted.

This is simpler than the originally proposed matrix, so it's not about simplifying code, it's about simplifying logic, and replacing decision tree with simpler linear decision sequence.

Reply:
so you don't need to negate anything, the code directly implements the required logic here